internal/cmd/perfcomp/main.go

```diff
@@ -0,0 +1,422 @@
+	// Check for variables
+	uri := os.Getenv("PERF_URI_PRIVATE_ENDPOINT")
+	if uri == "" {
+		log.Fatal("PERF_URI_PRIVATE_ENDPOINT env variable is not set")
+	}
+
+	version := os.Args[len(os.Args)-1]
+	if version == "" {
+		log.Fatal("could not get VERSION_ID")
+	}
+
+	// TODO (GODRIVER-3102): Map each project to a unique performance context,
+	// necessary for project switching to work since it's required for querying the stable region.
+	project := flag.String("project", "mongo-go-driver", "specify the name of an existing Evergreen project")
+	if project == nil {
+		log.Fatalf("must provide project")
+	}
+
+	// Connect to analytics node
+	client, err := mongo.Connect(options.Client().ApplyURI(uri))
+	if err != nil {
+		log.Fatalf("Error connecting client: %v", err)
+	}
+
+	defer func() { // Defer disconnect client
+		err = client.Disconnect(context.Background())
+		if err != nil {
+			log.Fatalf("Failed to disconnect client: %v", err)
+		}
+	}()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	err = client.Ping(ctx, nil)
+	if err != nil {
+		log.Fatalf("Error pinging MongoDB Analytics: %v", err)
+	}
+	log.Println("Successfully connected to MongoDB Analytics node.")
+
+	db := client.Database(expandedMetricsDB)
+
+	// Get raw data, most recent stable region, and calculate energy stats
+	findCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	patchRawData, err := findRawData(findCtx, *project, version, db.Collection(rawResultsColl))
+	if err != nil {
+		log.Fatalf("Error getting raw data: %v", err)
+	}
+
+	allEnergyStats, err := getEnergyStatsForAllBenchMarks(patchRawData, db.Collection(stableRegionsColl))
+	if err != nil {
+		log.Fatalf("Error getting energy statistics: %v", err)
+	}
+	log.Println(generatePRComment(allEnergyStats, version))
+}
+
+func findRawData(ctx context.Context, project string, version string, coll *mongo.Collection) ([]RawData, error) {
+	filter := bson.D{
+		{"info.project", project},
+		{"info.version", version},
+		{"info.variant", "perf"},
+		{"info.task_name", "perf"},
+	}
+
+	cursor, err := coll.Find(ctx, filter)
+	if err != nil {
+		log.Fatalf(
+			"Error retrieving raw data for version %q: %v",
+			version,
+			err,
+		)
+	}
+	defer func() {
+		err = cursor.Close(ctx)
+		if err != nil {
+			log.Fatalf("Error closing cursor while retrieving raw data for version %q: %v", version, err)
+		}
+	}()
+
+	log.Printf("Successfully retrieved %d docs from version %s.\n", cursor.RemainingBatchLength(), version)
+
+	var rawData []RawData
+	err = cursor.All(ctx, &rawData)
+	if err != nil {
+		log.Fatalf(
+			"Error decoding raw data from version %q: %v",
+			version,
+			err,
+		)
+	}
+
+	return rawData, err
+}
+
+// Find the most recent stable region of the mainline version for a specific test/measurement
+func findLastStableRegion(project string, testname string, measurement string, coll *mongo.Collection) (*StableRegion, error) {
```

Comment:
Suggest adding a context to the function signature, as this is a blocking function.